habitat/datasets/roomnav/roomnav_dataset.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import gzip
+import json
+import os
+from typing import List, Optional
+
+from habitat.config import Config
+from habitat.core.dataset import Dataset
+from habitat.core.registry import registry
+from habitat.tasks.nav.nav_task import (
+    RoomGoal,
+    RoomNavigationEpisode,
+    ShortestPathPoint,
+)
+
+ALL_SCENES_MASK = "*"
+CONTENT_SCENES_PATH_FIELD = "content_scenes_path"
+DEFAULT_SCENE_PATH_PREFIX = "data/scene_datasets/"
+
+
+@registry.register_dataset(name="RoomNav-v1")
+class RoomNavDatasetV1(Dataset):
```

Comment:
Maybe we can reduce code duplication if inherit from `PointGoalDatasetV1` or create common parent class.